x-pack/plugins/apm/public/hooks/use_fetcher.tsx

```diff
@@ -39,6 +42,14 @@ function getDetailsFromErrorResponse(error: IHttpFetchError) {
   );
 }
 
+const createLifecycleManagedCallApmApi = (
```

Comment:
nit (suggestion)
```suggestion
const createAbortableApmApiClient = (
```

Reply:
I considered this, but every ApmApiClient's calls are abortable, if you pass a signal. This one takes care of aborting for you. So I'm not sure if `createAbortableApmApiClient` is better here. But I agree that `createLifecycleManagedCallApmApi` is not great.

Reply:
Ah, okay. Would it make sense to call it `createAutoAbortableApmClient` or something like that? "lifecycle" is pretty very vague, and not something I'd immediately associate with the request being aborted automatically.

Reply:
Agreed, let me think about it, I think that's one better at least.